src/misc.ts

```diff
@@ -92,3 +92,87 @@ export const hasProperty = (
   object: RuntimeObject,
   name: string | number | symbol,
 ): boolean => Object.hasOwnProperty.call(object, name);
+
+export type PlainObject = Record<number | string | symbol, unknown>;
+
+/**
+ * Predefined sizes (in Bytes) of specific parts of JSON structure.
+ */
+export enum JsonSize {
+  NULL = 4,
+  COMMA = 1,
+  WRAPPER = 1,
+  TRUE = 4,
+  FALSE = 5,
+  ZERO = 1,
+  QUOTE = 1,
+  COLON = 1,
+  DOT = 1,
+}
+
+/**
+ * Check if the value is plain object.
+ *
+ * @param value - Value to be checked.
+ * @returns Boolean.
+ */
+export function isPlainObject(value: unknown): value is PlainObject {
+  if (typeof value !== 'object' || value === null) {
+    return false;
+  }
+
+  try {
+    let proto = value;
+    while (Object.getPrototypeOf(proto) !== null) {
+      proto = Object.getPrototypeOf(proto);
+    }
+
+    return Object.getPrototypeOf(value) === proto;
+  } catch (_) {
+    return false;
+  }
+}
+
+/**
+ * Check if character or string is ASCII.
+ *
+ * @param value - String or character.
+ * @returns Boolean, true if value is ASCII, false if not.
+ */
+export function isASCII(value: string) {
+  // eslint-disable-next-line no-control-regex,require-unicode-regexp
+  return /^[\x00-\x7F]*$/.test(value);
+}
+
+/**
+ * Calculate string size.
+ *
+ * @param value - String value to calculate size.
+ * @returns Number of bytes used to store whole string value.
+ */
+export function calculateStringSize(value: string) {
+  let size = 0;
+  for (const character of value) {
```

Comment:
Unfortunately not, because there is no `.reduce` on the value of type `string` in JavaScript, as far as I can see.

The only option would be turning that string into an array and then using reduce on it in some way. But, I doubt it would be any better. It would just add more redundant operations to the process.

Reply:
```ts
value.split('').reduce((total, character) => {
  if (isASCII(character)) {
    return total + 1;
  }
  return total + 2;
});
```

Reply:
Suggested code works well with little alteration (`initialValue` set to `0`). Refactored in the next commit. Thanks.